### DISCUSSION TITLE
Сolumns swap places in the prediction, PolynomialTrendForecaster

### DISCUSSION THREAD

**Discussion by eliza-petyaeva** [alan-turing-institute/sktime#4875] [Q&A] [answered]
Hi guys!
I found a bug while using PolynomialTrendForecaster on hierarchical data with multiple time series (columns). 
<img width="1142" alt="image" src="https://github.com/sktime/sktime/assets/72703346/1ff5f50b-9e3c-4d42-bb73-3086d663d587">
At the same time, everything is ok when columns' naming is just numbers - range[0, n]
<img width="1050" alt="image" src="https://github.com/sktime/sktime/assets/72703346/ce765a29-e242-4df3-8151-812db3be328f">

**Comment by fkiraly** [comment, marked as answer]
@eliza-petyaeva, this might be an instance of this bug
https://github.com/sktime/sktime/pull/4684
which was fixed in 0.19.2.

What would happen is, by accident, columns would get lexicographically ordered, but the column labels wouldn't.
In your case, `c17` is lexicographically before `c2`, so the bug happens.

The root cause was unexpected behaviour of `pandas` `pivot` (which sorted the columns in a case where you would not think).

Can you post your `sktime` version to check whether it's 0.19.1 or lower?
You can do this easily as follows:

```python
from sktime import show_versions

show_versions()
```
(and post the printout)

**Comment by eliza-petyaeva** [comment]
thx! All is fine 👍
_it was 0.16.1_
